src/olympia/reviewers/views.py

```diff
@@ -1313,9 +1315,60 @@ def list(self, request, *args, **kwargs):
     def retrieve(self, request, *args, **kwargs):
         serializer = AddonBrowseVersionSerializer(
             instance=self.get_object(),
-            context={'file': self.request.GET.get('file', None)})
+            context={
+                'file': self.request.GET.get('file', None),
+                'request': self.request
+            }
+        )
         return Response(serializer.data)
 
+    @action(
+        detail=True,
+        methods=['get'],
+        permission_classes=ReviewAddonVersionMixin.permission_classes,
+        url_path='download/(?P<filename>.*)')
+    def download(self, request, *args, **kwargs):
+        version = self.get_object()
+        file = version.current_file
+        file_param = kwargs['filename']
+
+        serializer = FileEntriesSerializer(
+            instance=file, context={
+                'file': file_param,
+                'request': request
+            }
+        )
+
+        commit = serializer._get_commit(file)
+        tree = serializer.repo.get_root_tree(commit)
+
+        try:
+            blob_or_tree = tree[serializer.get_selected_file(file)]
+
+            if blob_or_tree.type == 'tree':
+                return http.HttpResponseBadRequest('Can\'t serve directories')
+            selected_file = serializer.get_entries(file)[file_param]
+        except KeyError:
+            raise http.Http404()
+
+        actual_blob = serializer.git_repo[blob_or_tree.oid]
+        response = http.FileResponse(
+            memoryview(actual_blob),
+            content_type=selected_file['mimetype'])
+
+        # Backported from Django 2.1 to handle unicode filenames properly
```

Comment:
So is this something we can remove when we drop pre-2.2 support in the code?  Where is it backported from exactly?

Reply:
Correct, this has been backported from `django.http.response:FileResponse` - https://github.com/django/django/blob/edec11ce86a1a0d9e4c5a2a0df6acaf655041c24/django/http/response.py#L439